engine/common/api_helpers/mixins.py

```diff
@@ -141,8 +142,11 @@ def handle_exception(self, exc):
         return super().handle_exception(exc)
 
 
-class PublicPrimaryKeyMixin:
-    def get_object(self):
+_MT = typing.TypeVar("_MT", bound=models.Model)
+
+
+class PublicPrimaryKeyMixin(typing.Generic[_MT]):
+    def get_object(self) -> _MT:
```

Comment:
this is kinda cool. Here's [an example of how it's used](https://github.com/grafana/oncall/pull/2712/files#diff-0520dcc9ebfb340cb8390d8fc135fb45364c323abe5eb6bc508e140b58d38ac4R25). Basically it allows us to strongly type the return type of `get_object`.

For example, [here](https://github.com/grafana/oncall/pull/2712/files#diff-0520dcc9ebfb340cb8390d8fc135fb45364c323abe5eb6bc508e140b58d38ac4R96) my IDE now knows that `shift_swap` is of type `ShiftSwapRequest`.

Reply:
this will become muchhhh easier to do once Python 3.12 is released

https://docs.python.org/3.12/whatsnew/3.12.html#pep-695-type-parameter-syntax